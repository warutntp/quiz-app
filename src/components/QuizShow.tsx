import React from "react";

interface QuizShowProps {
  question: string;
  answers: string[];
  handleAnswer: (answer: string) => void;
  selectedAnswer: string | null;
}

const QuizShow: React.FC<QuizShowProps> = ({
  question,
  answers,
  handleAnswer,
  selectedAnswer,
}) => {
  return (
    <div className="p-4 border rounded-lg shadow-md bg-white">
      <h2 className="text-lg font-bold mb-4">{question}</h2>
      {answers.map((answer, index) => (
        <div
          key={index}
          className="px-2 mb-4 border rounded-lg shadow bg-blue-100 hover:bg-blue-300"
        >
          <label
            key={index}
            className="block my-2"
            aria-label={`Answer ${index + 1}`}
          >
            <input
              type="radio"
              name="answer"
              value={answer}
              checked={selectedAnswer === answer}
              onChange={() => handleAnswer(answer)}
              className="mr-2"
            />
            {answer}
          </label>
        </div>
      ))}
    </div>
  );
};

export default QuizShow;
