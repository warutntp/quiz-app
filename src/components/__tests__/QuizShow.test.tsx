import { render, screen, fireEvent } from "@testing-library/react";
import QuizShow from "../QuizShow";
import "@testing-library/jest-dom/extend-expect";

test("QuizShow", () => {
  const handleAnswer = jest.fn();
  const { container } = render(
    <QuizShow
      question="Sample Question?"
      answers={["Answer 1", "Answer 2"]}
      handleAnswer={handleAnswer}
      selectedAnswer={null}
    />
  );
  expect(container).toMatchSnapshot();
});

test("renders question and answers, selects answer correctly", () => {
  const handleAnswer = jest.fn();
  render(
    <QuizShow
      question="Sample Question?"
      answers={["Answer 1", "Answer 2"]}
      handleAnswer={handleAnswer}
      selectedAnswer={null}
    />
  );

  const answer1 = screen.getByLabelText(/Answer 1/i, { selector: "input" });
  fireEvent.click(answer1);
  expect(handleAnswer).toHaveBeenCalledWith("Answer 1");
});

test("renders question and answers with ARIA labels", () => {
  const handleAnswer = jest.fn();
  render(
    <QuizShow
      question="Sample Question?"
      answers={["Answer 1", "Answer 2"]}
      handleAnswer={handleAnswer}
      selectedAnswer={null}
    />
  );

  const answer1 = screen.getByLabelText(/Answer 1/i, { selector: "input" });
  expect(answer1).toBeInTheDocument();

  fireEvent.click(answer1);
  expect(handleAnswer).toHaveBeenCalledWith("Answer 1");
});
