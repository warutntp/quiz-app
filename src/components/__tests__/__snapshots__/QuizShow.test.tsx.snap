// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizShow 1`] = `
<div>
  <div
    class="p-4 border rounded-lg shadow-md bg-white"
  >
    <h2
      class="text-lg font-bold mb-4"
    >
      Sample Question?
    </h2>
    <div
      class="px-2 mb-4 border rounded-lg shadow bg-blue-100 hover:bg-blue-300"
    >
      <label
        aria-label="Answer 1"
        class="block my-2"
      >
        <input
          class="mr-2"
          name="answer"
          type="radio"
          value="Answer 1"
        />
        Answer 1
      </label>
    </div>
    <div
      class="px-2 mb-4 border rounded-lg shadow bg-blue-100 hover:bg-blue-300"
    >
      <label
        aria-label="Answer 2"
        class="block my-2"
      >
        <input
          class="mr-2"
          name="answer"
          type="radio"
          value="Answer 2"
        />
        Answer 2
      </label>
    </div>
  </div>
</div>
`;
